packages/bot-web-ui/src/components/dashboard/quick-strategy/quick-strategy-components/fields.tsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import classNames from 'classnames';
+import { SelectField, InputField, data_fields, data_uniq_input_obj } from '.';
+import { TDataFields } from './data/data-fields';
+import {
+    TSelectsFieldNames,
+    TDropdownItems,
+    TInputsFieldNames,
+    TDropdowns,
+    TSelectedValuesSelect,
+} from '../quick-strategy.types';
+import { TQuickStrategyFields } from './components.types';
+
+const QuickStrategyFields = React.memo(
+    ({
+        is_mobile,
+        types_strategies_dropdown,
+        symbol_dropdown,
+        trade_type_dropdown,
+        duration_unit_dropdown,
+        selected_type_strategy,
+        selected_trade_type,
+        selected_symbol,
+        selected_duration_unit,
+        onChangeDropdownItem,
+        onHideDropdownList,
+        setFieldValue,
+        onScrollStopDropdownList,
+        handleChange,
+        onChangeInputValue,
+        setCurrentFocus,
+        values,
+        description,
+        errors,
+    }: TQuickStrategyFields) => {
+        const is_double_idx_ref: React.MutableRefObject<number | undefined> = React.useRef();
+
+        const uniq_selected_input = React.useMemo(
+            () => data_uniq_input_obj.filter(elem => elem.index === selected_type_strategy.index)[0],
+            [selected_type_strategy]
+        );
+
+        const fields = React.useMemo(
+            () =>
+                data_fields.map((item, idx) => {
+                    const {
+                        id,
+                        field_name,
+                        className,
+                        label_className,
+                        field_className,
+                        input_value,
+                        select_value,
+                        label,
+                        placeholder,
+                        trailing_icon_message,
+                        zIndex,
+                        is_able_disabled,
+                        is_basic_field,
+                    } = item as TDataFields;
+
+                    const is_uniq_strategy_field = Array.isArray(item);
+                    const is_input_field = is_uniq_strategy_field || input_value;
+
+                    //dropdowns
+                    const types_strategies_options = id?.endsWith('types-strategies') && types_strategies_dropdown;
```

Comment:
```
id?.endsWith('types-strategies')
``` 
is used in different lines, can we create a variable for this instead.  

Also can this approach work? 
```
id === "types-strategies"
```

Reply:
@prince-deriv I left a comment to the same question, the code a little bit changed after this. 
`id === "types-strategies"` we can't use, there is a different value, but we can use it like in the first example as an alternative if you consider it as a better way
 ![Screenshot 2022-09-30 at 12 01 00](https://user-images.githubusercontent.com/103181650/193234350-80c3aedd-bd99-427a-9bf8-1d0aa0df7113.png)

Reply:
will be better
```
                    const type_strategy_pick = id === 'types-strategies' && selected_type_strategy;
                    const symbol_pick = id === 'symbol' && selected_symbol;
```

Reply:
fixed